framework/src/modules/chain/logic/transaction_pool.js

```diff
@@ -42,1058 +63,220 @@ const __private = {};
+			const transaction = transactions[i];
+			transactionList.push(transaction);
+		}
 
-/**
- * Adds a transaction to the multisignature index.
- *
- * @param {Object} transaction - Transaction object
- */
-TransactionPool.prototype.addMultisignatureTransaction = function(transaction) {
-	if (self.multisignature.index[transaction.id] === undefined) {
-		self.multisignature.transactions.push(transaction);
-		const index = self.multisignature.transactions.indexOf(transaction);
-		self.multisignature.index[transaction.id] = index;
-	}
-};
+		transactionList = reverse ? transactionList.reverse() : transactionList;
 
-/**
- * Removes a transaction from the multisignature index.
- *
- * @param {string} id - Transaction id
- */
-TransactionPool.prototype.removeMultisignatureTransaction = function(id) {
-	const index = self.multisignature.index[id];
+		if (limit) {
+			transactionList.splice(limit);
+		}
 
-	if (index !== undefined) {
-		self.multisignature.transactions[index] = false;
-		delete self.multisignature.index[id];
+		return transactionList;
 	}
-};
-
-/**
- * Counts the number of transactions in the multisignature index.
- *
- * @returns {number} Of transactions in the multisignature index
- */
-TransactionPool.prototype.countMultisignature = function() {
-	return Object.keys(self.multisignature.index).length;
-};
-
-/**
- * Receives transactions into the pool and add them to a queue.
- *
- * @param {Object[]} transactions - Array of received transactions
- * @param {boolean} broadcast - Broadcast flag
- * @param {function} cb - Callback function
- * @returns {SetImmediate} error, transactions
- */
-TransactionPool.prototype.receiveTransactions = function(
-	transactions,
-	broadcast,
-	cb
-) {
-	async.eachSeries(
-		transactions,
-		(transaction, eachSeriesCb) => {
-			self.processUnconfirmedTransaction(transaction, broadcast, eachSeriesCb);
-		},
-		err => setImmediate(cb, err, transactions)
-	);
-};
-
-/**
- * Reindexes all transaction queues, accounting for previously removed/falsified entries.
- */
-TransactionPool.prototype.reindexQueues = function() {
-	['bundled', 'queued', 'multisignature', 'unconfirmed'].forEach(queue => {
-		self[queue].index = {};
-		self[queue].transactions = self[queue].transactions.filter(Boolean);
-		self[queue].transactions.forEach(transaction => {
-			const index = self[queue].transactions.indexOf(transaction);
-			self[queue].index[transaction.id] = index;
-		});
-	});
-};
 
-/**
- * Processes the next bundle of transactions and add them to the queued index.
- *
- * @param {function} cb - Callback function
- * @returns {SetImmediate} error
- * @todo Compare / standardize the returns-description
- */
-TransactionPool.prototype.processBundled = function(cb) {
-	if (modules.loader.syncing()) {
-		// There is no need to process bundled transacions if node is syncing
-		return setImmediate(cb);
+	fillPool(cb) {
+		return this.pool
+			.processVerifiedTransactions()
+			.then(() => cb())
```

Comment:
No, we can not do it because `processVerifiedTransactions` returns a response. if we change the code to: `.then(cb)` then the `cb`'s first parameter - which is usually error - will have transactionsResponses and we do not want that.